src/components/ImageView/index.js

```diff
@@ -197,14 +203,15 @@ class ImageView extends PureComponent {
             return (
                 <View
                     style={[styles.imageViewContainer, styles.overflowHidden]}
+                    onLayout={this.onContainerLayoutChanged}
                 >
                     <Image
                         source={{uri: this.props.url}}
-                        style={[
+                        style={this.state.zoomScale === 0 ? undefined : [
                             styles.w100,
                             styles.h100,
-                        ]}
-                        resizeMode="contain"
+                        ]} // Hide image until zoomScale calculated
+                        resizeMode={this.state.zoomScale > 1 ? 'center' : 'contain'} // For big dimension images 'center' makes resizing the view. There is no sizing with 'contain'.
```

Comment:
I am not sure about this comment. What exactly do you mean by this?

Reply:
While doing my tests, I came across the following result. When big dimension photos come and we use center. Image component first shows the middle of the photo and then fits it to the screen. But we don't have this problem when we use 'contain'.

Reply:
Sorry, just saw the missing thing. Instead of `w100, h100` apply the image dimensions to the style when they gets available. Let's see if that fix it.

Reply:
Did you try this?

Reply:
Yes i tried it @parasharrajat , worked perfectly with small size photos. But it was not successful in large resolution photos. I used the following code. In the same way, I tried it by giving a value as "center" too . Unfortunately, it causes us to give a larger value than the container in large-sized photos.
```js
<View
                    style={[styles.imageViewContainer, styles.overflowHidden]}
                    onLayout={this.onContainerLayoutChanged}
                >
                    <Image
                        source={{uri: this.props.url}}
                        style={this.state.zoomScale === 0 ? undefined : [
                            {width: this.state.imgWidth, height: this.state.imgHeight},
                        ]} // Hide image until zoomScale calculated
                        resizeMode="contain"
                        onLoadStart={this.imageLoadingStart}
                        onLoadEnd={this.imageLoadingEnd}
                    />
                    {this.state.isLoading && (
                        <FullscreenLoadingIndicator
                            style={[styles.opacity1, styles.bgTransparent]}
                        />
                    )}
                </View>
```
video:

https://user-images.githubusercontent.com/6093207/173131949-3ea33978-0687-496b-897d-b1236ae569a9.mp4



--------
If you want to examine it, I have developed another solution method today. I hope I found a nice solution this time. 🙏 

```js
<View
                    style={[styles.imageViewContainer, styles.overflowHidden]}
                    onLayout={this.onContainerLayoutChanged}
                >
                    <Image
                        source={{uri: this.props.url}}
                        style={this.state.zoomScale === 0 ? undefined : [
                            this.state.zoomScale > 1 ? {width: this.state.imgWidth, height: this.state.imgHeight}
                                : {width: this.state.imgWidth * this.state.zoomScale, height: this.state.imgHeight * this.state.zoomScale},
                        ]} // Hide image until zoomScale calculated and multiplying the scale value by the resolution
                        resizeMode="contain"
                        onLoadStart={this.imageLoadingStart}
                        onLoadEnd={this.imageLoadingEnd}
                    />
                    {this.state.isLoading && (
                        <FullscreenLoadingIndicator
                            style={[styles.opacity1, styles.bgTransparent]}
                        />
                    )}
                </View>
```
video :

https://user-images.githubusercontent.com/6093207/173131981-91d25a62-4677-4ab6-9b7e-29ba32879356.mp4

Reply:
I see. Thanks. We are one step close. Now lets try `StyleUtils.getZoomSizingStyle` to apply the width and height. Let's see if that works.

Reply:
Thank you so much for this tips . Its look like this first time
![WhatsApp Image 2022-06-10 at 22 33 21](https://user-images.githubusercontent.com/6093207/173137534-d9bade19-eda6-4e12-8826-50f637a9fe13.jpeg)

But after adding position style looks perfect . 

Last version look like this😍.
```js
<View
                    style={[styles.imageViewContainer, styles.overflowHidden]}
                    onLayout={this.onContainerLayoutChanged}
                >
                    <Image
                        source={{uri: this.props.url}}
                        style={this.state.zoomScale === 0 ? undefined : [
                            StyleUtils.getZoomSizingStyle(this.state.isZoomed, this.state.imgWidth, this.state.imgHeight, this.state.zoomScale,
                                this.state.containerHeight, this.state.containerWidth),
                            styles.pAbsolute,
                        ]} // Hide image until zoomScale calculated
                        resizeMode="contain"
                        onLoadStart={this.imageLoadingStart}
                        onLoadEnd={this.imageLoadingEnd}
                    />
                    {this.state.isLoading && (
                        <FullscreenLoadingIndicator
                            style={[styles.opacity1, styles.bgTransparent]}
                        />
                    )}
                </View>
```
video :

https://user-images.githubusercontent.com/6093207/173138210-495071ce-d2e2-4695-a35e-39c70ae66d2f.mp4

Reply:
android works fine too btw  🥳

Reply:
Thanks, I will run a few tests to choose the best approach.